src/ui.rs

```diff
@@ -3047,40 +3047,60 @@ impl Component for SectionLineView<'_> {
     }
 
     fn draw(&self, viewport: &mut Viewport<Self::Id>, x: isize, y: isize) {
-        const NEWLINE_ICON: &str = "⏎";
-        let Self { line_key: _, inner } = self;
+        fn replacement_characters(character: char) -> Option<&'static str> {
+            match character {
+                // Characters end up writing over each-other and end up
+                // displaying incorrectly if ignored. Replacing tabs
+                // with a known length string fixes the issue for now.
+                '\t' => Some("→   "),
+                '\n' => Some("⏎"),
+                _ => None,
+            }
+        }
```

Comment:
note: I do think it would be desirable to parameterize the replacement strings via config at some point so that tabs aren't fixed width, but I'm not sure how you'd go about implementing that to get the context to this point.